collector/container/konflux.Dockerfile

```diff
@@ -0,0 +1,196 @@
+        diffutils \
+        elfutils-libelf-devel \
+        file \
+        flex \
+        gcc \
+        gcc-c++ \
+        gdb \
+        gettext \
+        git \
+        glibc-devel \
+        libasan \
+        libubsan \
+        libcap-ng-devel \
+        libcurl-devel \
+        libtool \
+        libuuid-devel \
+        make \
+        openssh-server \
+        openssl-devel \
+        patchutils \
+        passwd \
+        pkgconfig \
+        rsync \
+        tar \
+        unzip \
+        valgrind \
+        wget \
+        which \
+        bpftool \
+    && dnf clean all
+
+ARG BUILD_DIR
+ARG SRC_ROOT_DIR=${BUILD_DIR}
+ARG CMAKE_BUILD_DIR
+# TODO(ROX-20240): CMAKE_BUILD_TYPE should probably not be Release for PR, normal branch builds
+ARG CMAKE_BUILD_TYPE=Release
+# Appends an argument to the driver download URL that is used for filtering alerts on missing kernels.
+# TODO(ROX-20240): This needs to be true on PRs only.
+ARG COLLECTOR_APPEND_CID=false
+ARG USE_VALGRIND=false
+ARG ADDRESS_SANITIZER=false
+ARG TRACE_SINSP_EVENTS=false
+
+WORKDIR ${BUILD_DIR}
+
+RUN mkdir kernel-modules \
+    && cp -a /builder builder \
+    && cp -a /collector collector \
+    && cp -a /falcosecurity-libs falcosecurity-libs \
+    && cp -a /builder/third_party third_party \
+    && cp -a /kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION \
+    && cp -a /CMakeLists.txt CMakeLists.txt
+
+# WITH_RHEL_RPMS controls for dependency installation, ie if they were already installed as RPMs.
+# Setting the value to empty will cause dependencies to be downloaded from repositories or accessed in submodules and compiled.
+# TODO(ROX-20651): Set ENV WITH_RHEL_RPMS=true when RHEL RPMs can be installed to enable hermetic builds.
+# ENV WITH_RHEL_RPMS=true
+
+# Build with gperftools (DISABLE_PROFILING=OFF) only for supported
+# architectures, at the moment x86_64 only
+RUN ./builder/install/install-dependencies.sh && \
+    if [[ "$(uname -m)" == "x86_64" ]];   \
+        then DISABLE_PROFILING="OFF";   \
+        else DISABLE_PROFILING="ON";    \
+    fi ; \
+    cmake -S ${SRC_ROOT_DIR} -B ${CMAKE_BUILD_DIR} \
+           -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+           -DDISABLE_PROFILING=${DISABLE_PROFILING} \
+           -DCOLLECTOR_APPEND_CID=${COLLECTOR_APPEND_CID} \
+           -DUSE_VALGRIND=${USE_VALGRIND} \
+           -DADDRESS_SANITIZER=${ADDRESS_SANITIZER} \
+           -DTRACE_SINSP_EVENTS=${TRACE_SINSP_EVENTS} && \
+    cmake --build ${CMAKE_BUILD_DIR} --target all -- -j "${NPROCS:-2}" && \
+    ctest -V --test-dir ${CMAKE_BUILD_DIR} && \
+    strip -v --strip-unneeded "${CMAKE_BUILD_DIR}/collector/collector"
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS support-packages-downloader
+
+WORKDIR /staging
+
+COPY kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION
+COPY collector/container/scripts/download-support-package.sh download-support-package.sh
+
+RUN ./download-support-package.sh
```

Comment:
I think, the amount of effort to make this script run in a Tekton stage (see example [here](https://github.com/stackrox/scanner/blob/b29219d34d7ccff433aadfe7807e71b79b0830f8/.tekton/scanner-pull-request.yaml#L236-L254)) is the same as in Dockerfile.

The advantage of the Tekton stage is that we don't need to modify this part of Dockerfile when switching on hermetic builds. Since, I hope, the amount of effort to change isn't large, I suggest to make it as part of this PR.